nixos/modules/tasks/network-interfaces-systemd.nix

```diff
@@ -59,7 +59,14 @@ in
           in {
             DHCP = override (dhcpStr cfg.useDHCP);
           } // optionalAttrs (gateway != [ ]) {
-            gateway = override gateway;
+            routes = override [
```

Comment:
I am not convinced this is the best way to solve this right now.

The issue was that we would add `onlink` routes for all network devices on the system. Introduced in https://github.com/systemd/systemd/commit/4912ab7774ccc1eed07bfbbc6b87e8a23b1f5495 systemd would set the `onlink` attribute for all routes that didn't come with static addresses.

Reply:
It's certainly not ideal but then again adding the default gateway to every interface is FUBAR anyway. This fix is IMHO fine until we finally deprecate `networking.defaultGateway` and move it to the per interface configurations.

Reply:
:+1: I created an issue for this (#61629) and also added it to our systemd project board https://github.com/NixOS/nixpkgs/projects/22. (I am trying to populate that more with all the things we come across wile working on this PR).